contracts/multisigs/GnosisSafeSameAddressMultisig.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+// Gnosis Safe Master Copy interface extracted from the mainnet: https://etherscan.io/address/0xd9db270c1b5e3bd161e8c8503c55ceabee709552#code#F6#L126
+interface IGnosisSafe {
+    /// @dev Gets set of owners.
+    /// @return Set of Safe owners.
+    function getOwners() external view returns (address[] memory);
+
+    /// @dev Gets threshold.
+    /// @return Threshold
+    function getThreshold() external view returns (uint256);
+}
+
+/// @dev Provided incorrect data length.
+/// @param expected Expected minimum data length.
+/// @param provided Provided data length.
+error IncorrectDataLength(uint256 expected, uint256 provided);
+
+/// @dev Provided incorrect multisig threshold.
+/// @param expected Expected threshold.
+/// @param provided Provided threshold.
+error WrongThreshold(uint256 expected, uint256 provided);
+
+/// @dev Provided incorrect number of owners.
+/// @param expected Expected number of owners.
+/// @param provided Provided number of owners.
+error WrongNumOwners(uint256 expected, uint256 provided);
+
+/// @dev Provided incorrect multisig owner.
+/// @param provided Provided owner address.
+error WrongOwner(address provided);
+
+/// @title Gnosis Safe Same Address - Smart contract for Gnosis Safe verification of an already existent multisig address.
+/// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
+contract GnosisSafeSameAddressMultisig {
+    // Default data size to be parsed as an address of a Gnosis Safe multisig
+    uint256 public constant DEFAULT_DATA_LENGTH = 20;
```

Comment:
All we need is the address of a multisig that was already deployed

Reply:
Updated that part. Actually, the master copy address is not needed at all, we don't use it except for the interface